src/views/Leads/Table/index.js

```diff
@@ -108,19 +89,105 @@ const requestOptions = {
 
-@connect(mapStateToProps, mapDispatchToProps)
-@RequestClient(requestOptions)
-export default class Table extends React.Component {
-    static propTypes = propTypes;
-    static defaultProps = defaultProps;
-    static leadKeyExtractor = lead => String(lead.id)
-
-    constructor(props) {
-        super(props);
-
-        const { headersMap } = this.props;
+const leadKeyExtractor = lead => String(lead.id);
+
+function Table(props) {
+    const {
+        headersMap,
+        leads,
+        emptyComponent,
+        loading,
+        isFilterEmpty,
+        className,
+        requests: {
+            leadPatchRequest: { pending },
+        },
+        onSearchSimilarLead,
+        onRemoveLead,
+        activeProject,
+    } = props;
+
+    const {
+        values: selectedLeads,
+        isItemPresent,
+        clickOnItem,
+        addItems,
+        removeItems,
+        clearSelection,
+    } = useArraySelection(
+        leadKeyExtractor,
+        [],
+    );
+
+    const handleClickItem = useCallback(
+        (newValue) => {
+            clickOnItem(newValue);
+        }, [clickOnItem],
+    );
+
+    const handleSelectAllCheckboxClick = useCallback(
+        (newValue) => {
+            if (newValue) {
+                addItems(leads);
+            } else {
+                removeItems(leads);
+            }
+        }, [leads, addItems, removeItems],
+    );
+
+    const {
+        areAllChecked,
+        areSomeChecked,
+    } = useMemo(() => {
+        if (!leads || leads.length === 0) {
+            return {
+                areAllChecked: false,
+                areSomeChecked: false,
+            };
+        }
+        const filteredLeads = leads.filter(l => isItemPresent(leadKeyExtractor(l)));
+        return {
+            areAllChecked: filteredLeads.length === leads.length,
+            areSomeChecked: filteredLeads.length < leads.length && filteredLeads.length > 0,
+        };
+    }, [leads, isItemPresent]);
+
+    const headers = useMemo(
+        () => [
+            {
+                key: 'multi_select',
+                order: 0,
+                modifier: (row) => {
+                    const itemSelected = selectedLeads.length > 0 && (
+                        selectedLeads.find(lead => lead.id === row.id)
+                    );
 
-        const headers = [
+                    return (
+                        <Checkbox
+                            onChange={() => handleClickItem(row)}
+                            value={!!itemSelected}
+                            className={styles.checkbox}
+                        />
+                    );
+                },
+                label: areAllChecked ? (
+                    <Icon
+                        onClick={clearSelection} // TODO: styling of button
+                        name="minusOutline"
+                        transparent
+                        className={styles.clearIcon}
+                    />
```

Comment:
Let's not do this.
If all are checked: a checkmark should appear
If none of them are checked, an empty box should appear
If some of them are checked, an intermediate state is shown (this is optional, lets not do this for now)

So, only the checkbox will solve this issue, we don't need this `Icon` here.